plugins/org.jkiss.dbeaver.model.sm/src/org/jkiss/dbeaver/model/security/SMController.java

```diff
@@ -230,4 +230,31 @@ void deleteAllObjectPermissions(
 
     @NotNull
     String[] getTeamMembers(String teamId) throws DBException;
+
+    /**
+     * Decodes token and gets user id from token.
+     */
+    @NotNull
+    String getUserIdFromAPIToken(@NotNull String token) throws DBException;
+
+    /**
+     * Returns all info about user API tokens that stored in database.
+     */
+    @NotNull
+    List<SMAPIToken> getAPITokens() throws DBException;
+
+    /**
+     * Creates API token for a user.
+     *
+     * @param tokenName name of a token
+     * @param period    period of validity
+     * @return API token info (including a private key)
+     */
+    @NotNull
+    SMAPIToken createAPIToken(@NotNull String tokenName, int period) throws DBException;
+
+    /**
+     * Removes info about API token for a user.
+     */
+    void deleteAPIToken(@NotNull String tokenName) throws DBException;
```

Comment:
```suggestion
    void deleteAPIToken(@NotNull String tokenId) throws DBException;
```

Reply:
or we really want to delete token by name?

Reply:
we need to delete token by name